vault/identity_store_oidc_provider.go

```diff
@@ -0,0 +1,164 @@
+package vault
+
+import (
+	"context"
+
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+type assignment struct {
+	name     string
```

Comment:
It doesn't look like this gets referenced anywhere. The name is part of the storage entry so we don't need to keep this field around.